chainntnfs/txconfnotifier.go

```diff
@@ -193,13 +219,51 @@ func (tcn *TxConfNotifier) ConnectTip(blockHash *chainhash.Hash,
 			}
 			ntfnSet[ntfn] = struct{}{}
 
-			tcn.confTxsByInitialHeight[blockHeight] =
-				append(tcn.confTxsByInitialHeight[blockHeight], tx.Hash())
+			txSet, exists := tcn.txsByInitialHeight[blockHeight]
+			if !exists {
+				txSet = make(map[chainhash.Hash]struct{})
+				tcn.txsByInitialHeight[blockHeight] = txSet
+			}
+			txSet[*txHash] = struct{}{}
+		}
+	}
+
+	// Next, we'll dispatch an update to all of the notification clients for
+	// our watched transactions with the number of confirmations left at
+	// this new height.
+	for _, txHashes := range tcn.txsByInitialHeight {
+		for txHash := range txHashes {
+			for _, ntfn := range tcn.confNotifications[txHash] {
+				// If the transaction still hasn't been included
+				// in a block, we'll skip it.
+				if ntfn.details == nil {
+					continue
+				}
+
+				txConfHeight := ntfn.details.BlockHeight +
```

Comment:
Why's the conf height not just based on the first value in the map `txsByInitialHeight`, assuming that variable is captured at `firstConfHeight`, then this would be something like:
```
txConfHeight := ntfn.details.BlockHeight - firstConfHeight - 1
``

Reply:
Not sure what you mean here. `ntfn.details.BlockHeight` is the height of the block the transaction was included in, which is analogous to the height of the first confirmation. Going by that, the above would always result in -1, no?